x-pack/plugins/osquery/public/routes/saved_queries/index.tsx

```diff
@@ -12,18 +12,29 @@ import { QueriesPage } from './list';
 import { NewSavedQueryPage } from './new';
 import { EditSavedQueryPage } from './edit';
 import { useBreadcrumbs } from '../../common/hooks/use_breadcrumbs';
+import { MissingPrivileges } from '../components';
+import { useKibana } from '../../common/lib/kibana';
 
 const SavedQueriesComponent = () => {
+  const {
+    allSavedQueries,
+    readSavedQueries,
+    runSavedQueries,
+  } = useKibana().services.application.capabilities.osquery;
   useBreadcrumbs('saved_queries');
   const match = useRouteMatch();
 
+  if (!readSavedQueries) {
+    return <MissingPrivileges />;
+  }
+
   return (
     <Switch>
       <Route path={`${match.url}/new`}>
-        <NewSavedQueryPage />
+        {allSavedQueries || runSavedQueries ? <NewSavedQueryPage /> : <MissingPrivileges />}
       </Route>
       <Route path={`${match.url}/:savedQueryId`}>
-        <EditSavedQueryPage />
+        <EditSavedQueryPage viewMode={!allSavedQueries} />
       </Route>
```

Comment:
1. As we have discussed with the team there may be a case when a user has read privileges for saved queries and live queries, so it does mean he can just preview the results of live queries and see what queries are available, but without actually being able to run any of them (agree that maybe right now it may be overkill, but it's just base RBAC that we're going to extend soon, for example with the option to be able to run Osquery from Metrics UI https://github.com/elastic/kibana/pull/104272 or from the Cases view, so we're not 100% sure yet how the RBAC will evolve, so we just wanted to make it flexible)
2. You're right, I need to fix that 👍 
3. Makes sense, I'm going to change that 👍